test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -8300,7 +8300,7 @@ BOOST_AUTO_TEST_CASE(inline_assembly_memory_access)
 			function test() returns (bytes) {
 				bytes memory x = new bytes(5);
 				for (uint i = 0; i < x.length; ++i)
-					x[i] = byte(i + 1);
+					x[i] = byte(uint8(i + 1));
```

Comment:
Should this generate an error? - @axic

Reply:
`i` is of type `uint`, therefore so is `i+1`, so it cannot be directly converted to `byte`.